src/compiler/transformers/ts.ts

```diff
@@ -2176,21 +2188,27 @@ namespace ts {
          * @param node The parameter declaration node.
          */
         function visitParameter(node: ParameterDeclaration) {
-            if (node.name && (node.name as Identifier).originalKeywordKind === SyntaxKind.ThisKeyword) {
+            if (node.name && isIdentifier(node.name) && (node.name as Identifier).originalKeywordKind === SyntaxKind.ThisKeyword) {

```

Comment:
I don't think you need to do this `(node.name as Identifier)` since it should already be type guarded